packages/amplify_api/lib/amplify_api.dart

```diff
@@ -33,7 +33,7 @@ class AmplifyAPI extends APIPluginInterface {
 
   static set instance(APIPluginInterface instance) {
     PlatformInterface.verifyToken(instance, _token);
-    _instance = instance;
+    _instance = instance as AmplifyAPI;
```

Comment:
Same as the other NS PRs, I'm not sure why the getters/setters are of type `APIPluginInterface` and not `AmplifyAPI`. I left more detailed comments on the other PRs.

Reply:
I had same question.

Reply:
changed instance type to be APIPlugin